src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagService.cs

```diff
@@ -79,14 +110,14 @@ public async Task<AddCustomTagResponse> AddCustomTagAsync(IEnumerable<CustomTagE
             {
                 // Reindex from latest one to earliest
                 // Current solution won't be able to handle when reindex fail in the middle, when moving to job framework, should solve it.
-                await _reindexJob.ReindexAsync(addedTags.Values, lastWatermark.Value);
+                await _reindexJob.ReindexAsync(addedCustomTagStoreEntries, lastWatermark.Value);
```

Comment:
What happens if Reindexing fails?

Reply:
Job framework will not handle it. It just executes the task you give to it. If it fails it is app failure like dicom tag value parsing fails, it will return failed status to reindex job

Reply:
I suppose the job  should have status like Thread, so that when job fails, it should allow resume based on the status.